src/request-id.js

```diff
@@ -0,0 +1,27 @@
+let requestId = null;
+
+/**
+ * Express.js middleware
+ * responsible for retrieving and saving the request id
+ * @return {function} express middleware
+ **/
+function middleware() {
+  return (req, res, next) => {
+    requestId = req.header('X-Request-ID') || null;
```

Comment:
I don't think this is safe, as another request might be handled concurrently when winston transport is async, before you `get()` the requestId and it might then be overridden. Also `logger.error` might not be called in the same request context or any for that matter. If you want to log additional error information, can't that be done in the express error handler middleware https://github.com/spring-media/themis-graphql/blob/lean-integration-on-connect-async/src/server.js#L172?

Reply:
Thanks for the suggestion! I did not write this code, I just ported it into this branch. Hopefully @veitbjarsch can take over.

Reply:
The idea is to configure the logger per request, i.e. add the content of a given HTTP request header to each log message that is issued throughout the lifetime of that request.

Reply:
Possible, but would need tighter integration of the logger with a wrapper adding additional meta info for log messages.